File: app/src/main/java/com/huck/biblequiz/data/local/entity/ChapterProgressEntity.kt
package com.huck.biblequiz.data.local.entity

import androidx.room.Entity

@Entity(tableName = "chapter_progress", primaryKeys = ["bookId", "chapter"])
data class ChapterProgressEntity(
    val bookId: Int,
    val chapter: Int,
    val studied: Boolean = false,
    val quizzed: Boolean = false,
    val bestScore: Int = 0,
    val totalQuestions: Int = 0
)

File: ios/BeSpecific/Data/Local/ChapterProgressEntity.swift
import Foundation
import SwiftData

@Model
final class ChapterProgressEntity {
    var bookId: Int
    var chapter: Int
    var studied: Bool
    var quizzed: Bool
    var bestScore: Int
    var totalQuestions: Int

    init(bookId: Int, chapter: Int, studied: Bool = false, quizzed: Bool = false, bestScore: Int = 0, totalQuestions: Int = 0) {
        self.bookId = bookId
        self.chapter = chapter
        self.studied = studied
        self.quizzed = quizzed
        self.bestScore = bestScore
        self.totalQuestions = totalQuestions
    }
}
